### ISSUE TITLE
bug(mat-calendar): Add tooltips to previous/next buttons in the header

### ISSUE THREAD

**Issue by PolinaGo** [angular/components#31536]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

The mat-calendar-header (src/material/datepicker/calendar-header.html) should have _tooltips_ on the button controls, specifically "Previous month", "Next month" to satisfy a11y: https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html. Note that these buttons already have corresponding aria-labels.

### Reproduction

StackBlitz link: https://stackblitz.com/run?file=src%2Fexample%2Fdatepicker-overview-example.html
Steps to reproduce:
1. Hover over the active previous or next button of the mat calendar header.
2. Observe the lack of tooltip

### Expected Behavior

Expect to have "Previous {period}" tooltip on the active "<" button and "Next {period}" tooltip on the active ">" button of the calendar header

### Actual Behavior

No tooltips, only aria-labels.

### Environment

Irrelevant
